solidity/contracts/system/TBTCSystem.sol

```diff
@@ -590,23 +594,58 @@ contract TBTCSystem is Ownable, ITBTCSystem, DepositLog {
     }
 
     /// @notice Request a new keep opening.
-    /// @param _m Minimum number of honest keep members required to sign.
-    /// @param _n Number of members in the keep.
+    /// @param _lotSizeSatoshis Lot size in satoshis.
     /// @param _maxSecuredLifetime Duration of stake lock in seconds.
     /// @return Address of a new keep.
     function requestNewKeep(
-        uint256 _m,
-        uint256 _n,
-        uint256 _bond,
+        uint64 _lotSizeSatoshis,
         uint256 _maxSecuredLifetime
     )
         external
         payable
         returns (address)
     {
         require(tbtcDepositToken.exists(uint256(msg.sender)), "Caller must be a Deposit contract");
+        require(isAllowedLotSize(_lotSizeSatoshis), "provided lot size not supported");
+
         IBondedECDSAKeepFactory _keepFactory = keepFactorySelection.selectFactoryAndRefresh();
-        return _keepFactory.openKeep.value(msg.value)(_n, _m, msg.sender, _bond, _maxSecuredLifetime);
+        uint256 bond = calculateBondRequirementWei(_lotSizeSatoshis);
+        return _keepFactory.openKeep.value(msg.value)(keepSize, keepThreshold, msg.sender, bond, _maxSecuredLifetime);
+    }
+
+    /// @notice Calculates bond requirement in wei for the given lot size in
+    /// satoshis given the current BTCETH price.
+    /// @param _lotSizeSatoshis Lot size in satoshis.
+    /// @return Bond requirement in wei.
+    function calculateBondRequirementWei(
+        uint256 _lotSizeSatoshis
+    ) public view returns (uint256) {
```

Comment:
I think this needs to stay `internal` or we're leaking the price info.

Reply:
Ah, good point. Will update it.